File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/global/fetchoptions/GlobalSearchObjectSortOptions.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.global.fetchoptions;

import com.fasterxml.jackson.annotation.JsonIgnore;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.SortOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.SortOrder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.global.GlobalSearchObject;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.global.fetchoptions.GlobalSearchObjectSortOptions")
public class GlobalSearchObjectSortOptions extends SortOptions<GlobalSearchObject>
{

    private static final long serialVersionUID = 1L;

    @JsonIgnore
    public static final String SCORE = "SCORE";

    @JsonIgnore
    public static final String OBJECT_KIND = "OBJECT_KIND";

    @JsonIgnore
    public static final String OBJECT_PERM_ID = "OBJECT_PERM_ID";

    @JsonIgnore
    public static final String OBJECT_IDENTIFIER = "OBJECT_IDENTIFIER";

    public SortOrder score()
    {
        return getOrCreateSorting(SCORE);
    }

    public SortOrder objectKind()
    {
        return getOrCreateSorting(OBJECT_KIND);
    }

    public SortOrder objectPermId()
    {
        return getOrCreateSorting(OBJECT_PERM_ID);
    }

    public SortOrder objectIdentifier()
    {
        return getOrCreateSorting(OBJECT_IDENTIFIER);
    }

}

File: plasmid/source/core-plugins/eln-lims/1/as/webapps/eln-lims/html/lib/openbis/js/v3/as/dto/global/fetchoptions/GlobalSearchObjectSortOptions.js
define([ "require", "stjs", "as/dto/common/fetchoptions/SortOptions" ], function(require, stjs, SortOptions) {
	var GlobalSearchObjectSortOptions = function() {
		SortOptions.call(this);
	};

	var fields = {
		SCORE : "SCORE",
		OBJECT_KIND : "OBJECT_KIND",
		OBJECT_PERM_ID : "OBJECT_PERM_ID",
		OBJECT_IDENTIFIER : "OBJECT_IDENTIFIER"
	};

	stjs.extend(GlobalSearchObjectSortOptions, SortOptions, [ SortOptions ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.global.fetchoptions.GlobalSearchObjectSortOptions';
		constructor.serialVersionUID = 1;
		prototype.score = function() {
			return this.getOrCreateSorting(fields.SCORE);
		};
		prototype.getScore = function() {
			return this.getSorting(fields.SCORE);
		};
		prototype.objectKind = function() {
			return this.getOrCreateSorting(fields.OBJECT_KIND);
		};
		prototype.getObjectKind = function() {
			return this.getSorting(fields.OBJECT_KIND);
		};
		prototype.objectPermId = function() {
			return this.getOrCreateSorting(fields.OBJECT_PERM_ID);
		};
		prototype.getObjectPermId = function() {
			return this.getSorting(fields.OBJECT_PERM_ID);
		};
		prototype.objectIdentifier = function() {
			return this.getOrCreateSorting(fields.OBJECT_IDENTIFIER);
		};
		prototype.getObjectIdentifier = function() {
			return this.getSorting(fields.OBJECT_IDENTIFIER);
		};
	}, {});
	return GlobalSearchObjectSortOptions;
})